pkg/sql/parser/scan.go

```diff
@@ -620,6 +620,15 @@ func (s *Scanner) scanNumber(lval *sqlSymType, ch int) {
 			return
 		}
 
+		// Strip off leading zeros from decimal literals so that
+		// constant.MakeFromLiteral doesn't inappropriately interpret the
+		// string as an octal literal.
+		if !isHex {
+			for len(lval.str) > 1 && lval.str[0] == '0' {
```

Comment:
Yup, and `strings.TrimLeft(lval.str, '0')` would be even nicer. Anyway, added a comment!